integration-tests/src/test/java/org/openmrs/module/fhir2/providers/r3/BaseFhirR3IntegrationTest.java

```diff
@@ -112,4 +117,98 @@ protected void describeMismatchSafely(Bundle.BundleEntryComponent item, Descript
 			matcher.describeMismatch(item.getResource(), mismatchDescription);
 		}
 	}
+	
+	protected static Matcher<List<Bundle.BundleEntryComponent>> isSortedAndWithinMax(Integer max) {
+		return new IsSortedAndWithinMax(max);
+	}
+	
+	private static class IsSortedAndWithinMax extends TypeSafeDiagnosingMatcher<List<Bundle.BundleEntryComponent>> {
+		
+		private int max;
+		
+		IsSortedAndWithinMax(int max) {
+			this.max = max;
+		}
+		
+		@Override
+		protected boolean matchesSafely(List<Bundle.BundleEntryComponent> entries, Description mismatchDescription) {
+			List<Observation> observations = entries.stream().map(Bundle.BundleEntryComponent::getResource)
+			        .filter(it -> it instanceof Observation).map(it -> (Observation) it).collect(Collectors.toList());
+			
+			Set<String> codeList = new HashSet<>();
+			
+			for (int var = 0; var < observations.size(); var++) {
+				String currentConcept = observations.get(var).getCode().getCodingFirstRep().getCode();
+				
+				//check if Bundle is returned grouped code wise
+				// if an observation arrives whose concept wise list is already created, then that means this observation is
+				//out of place
+				//so bundle is not returned grouped wise
+				if (codeList.contains(currentConcept)) {
+					mismatchDescription.appendText("an Observation which was not grouped correctly")
+					        .appendValue(observations.get(var).getId());
+					return false;
+				} else {
+					
+					codeList.add(currentConcept);
+					
+					String currentDateTimeType = observations.get(var).getEffectiveDateTimeType().toString();
+					
+					int distinctObsDateTime = 1;
+					
+					if (var == observations.size() - 1) {
+						return true;
+					}
+					
+					String nextConcept = observations.get(var + 1).getCode().getCodingFirstRep().getCode();
+					String nextDateTimeType = observations.get(var + 1).getEffectiveDateTimeType().toString();
+					
+					while (nextConcept.equals(currentConcept)) {
+						//if nextDatetime is greater than currentDateTime, then the list is not sorted in decreasing order, which was required
+						if (currentDateTimeType.compareTo(nextDateTimeType) < 0) {
+							mismatchDescription.appendText("an Observation which is not placed in sorted order")
+							        .appendValue(observations.get(var + 1).getId());
```

Comment:
I would include the date that we expected the observation to be greater than here.